core/src/transfer.rs

```diff
@@ -292,6 +293,48 @@ impl Transaction {
         }
     }
 
+    /// Returns the Blob used with the transaction, if any.
+    #[must_use]
+    pub fn blob(&self) -> Option<&Vec<BlobData>> {
+        match self {
+            Self::Phoenix(tx) => tx.blob(),
+            Self::Moonlight(tx) => tx.blob(),
+        }
+    }
+
+    /// Remove the blobs data from the transaction, and return them associated
+    /// to their blob hash
+    #[must_use]
+    pub fn take_blobs_data(&mut self) -> Option<Vec<([u8; 32], BlobDataPart)>> {
+        let blob = match self {
+            Self::Phoenix(tx) => tx.blob_mut(),
+            Self::Moonlight(tx) => tx.blob_mut(),
+        }?;
+
+        let ret = blob
+            .iter_mut()
+            .filter_map(|b| b.take_data().map(|d| (b.hash, d)))
+            .collect::<Vec<_>>();
+
+        Some(ret)
+    }
+
+    /// Creates a modified clone of this transaction if it contains a Blob,
+    /// clones all fields except for the Blob, where its hash is set as Memo.
+    ///
+    /// Returns none if the transaction is not a Blob transaction.
+    #[must_use]
+    pub fn blob_to_memo(&self) -> Option<Self> {
```

Comment:
Wouldn't it save you a lot of trouble down the line if this would just return the unmodified tx if there is not blob?

Reply:
It's a matter of references

If I return the original tx, I need to take the ownership cause I cannot return a reference of the new cloned/modified